### PR TITLE
SMP v3: encrypt message timestamp and flags together with the body between server and recipient

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: protocol/simplex-messaging.md [hunk 1]
```diff
@@ -360,6 +360,8 @@ The clients can optionally instruct a dedicated push notification server to subs
 - `subscribeNotifications` (`"NSUB"`) - see [Subscribe to queue notifications](#subscribe-to-queue-notifications).
 - `messageNotification` (`"NMSG"`) - see [Deliver message notification](#deliver-message-notification).
 
+[`SEND` command](#send-message) includes the notification flag to instruct SMP server whether to send the notification - this flag is forwarded to the recepient inside encrypted envelope, together with the timestamp and the message body, so even if TLS is compromised this flag cannot be used for traffic correlation.
```
Comment (jr-simplex) [inline protocol/simplex-messaging.md:363]: meta is outside encrypted envelope in `messageWithMeta` variant of `message` below - maybe it's better to mention here it's old protocol or in`MSG` abnf it's for backwards compatibility

File: tests/AgentTests/NotificationTests.hs [hunk 2]
```diff
@@ -268,26 +268,39 @@ testNotificationSubscriptionNewConnection APNSMockServer {apnsQ} = do
     (bobId, qInfo) <- createConnection alice SCMInvitation
     liftIO $ threadDelay 500000
     aliceId <- joinConnection bob qInfo "bob's connInfo"
+    liftIO $ print 1
```
Comment (jr-simplex) [inline tests/AgentTests/NotificationTests.hs:271]: just as a reminder to remove prints before merging

Comment (epoberezkin) [reply tests/AgentTests/NotificationTests.hs:271]: I am debugging the failing test

File: src/Simplex/Messaging/Server.hs [hunk 3]
```diff
@@ -569,36 +579,56 @@ client clnt@Client {subscriptions, ntfSubscriptions, rcvQ, sndQ} Server {subscri
...
+        encryptMsg qr Message {msgId, msgTs, msgFlags, msgBody} =
+          case thVersion of
+            3 -> encrypt $ encodeRcvMsgBody RcvMsgBody {msgTs, msgFlags, msgBody}
+            _ -> encrypt msgBody
+          where
+            encrypt :: KnownNat i => C.MaxLenBS i -> RcvMessage
+            encrypt body =
+              let encBody = EncRcvMsgBody $ C.cbEncryptMaxLenBS (rcvDhSecret qr) (C.cbNonce msgId) body
+               in RcvMessage msgId msgTs msgFlags encBody
+        -- 3 ->
```
Comment (jr-simplex) [inline src/Simplex/Messaging/Server.hs:625]: comment

File: protocol/simplex-messaging.md (hunk 1 above)
Comment (epoberezkin) [reply protocol/simplex-messaging.md:363]: updated

**Pull request merged**